includes/class-amp-story-post-type.php

```diff
@@ -1798,6 +1799,99 @@ public static function render_block_latest_stories( $attributes ) {
 		return ob_get_clean();
 	}
 
+	/**
+	 * Registers the Page Attachment block.
+	 */
+	public static function register_block_page_attachment() {
+		register_block_type(
+			'amp/amp-story-page-attachment',
+			[
+				'attributes'      => [
+					'postId'          => [
+						'type' => 'number',
+					],
+					'title'           => [
+						'type'    => 'string',
+						'default' => '',
+					],
+					'text'            => [
+						'type'    => 'string',
+						'default' => __( 'Swipe up', 'amp' ),
+					],
+					'theme'           => [
+						'type'    => 'string',
+						'default' => 'light',
+					],
+					'wrapperStyle'    => [
+						'type'    => 'object',
+						'default' => [],
+					],
+					'attachmentClass' => [
+						'type'    => 'string',
+						'default' => 'amp-page-attachment-content',
+					],
+				],
+				'render_callback' => [ __CLASS__, 'render_block_page_attachment' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the dynamic block Page Attachment.
+	 *
+	 * @param array $attributes The block attributes.
+	 * @return string $markup The rendered block markup.
+	 */
+	public static function render_block_page_attachment( $attributes ) {
+		global $post;
+
+		if ( empty( $attributes['postId'] ) ) {
+			return null;
+		}
+
+		$content_post = get_post( absint( $attributes['postId'] ) );
+
+		if ( empty( $content_post ) ) {
+			return null;
+		}
+
+		$original_post = $post;
+		$post          = $content_post; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		// Remove filter for not adding grid layer to blocks within the attachment content.
+		remove_filter( 'render_block', [ __CLASS__, 'render_block_with_grid_layer' ], 10 );
+
+		setup_postdata( $content_post );
+
+		$style = '';
+		if ( isset( $attributes['wrapperStyle']['backgroundColor'] ) ) {
+			$style .= 'background-color:' . $attributes['wrapperStyle']['backgroundColor'] . ';';
+		}
+		if ( isset( $attributes['wrapperStyle']['color'] ) ) {
+			$style .= 'color:' . $attributes['wrapperStyle']['color'] . ';';
+		}
+
+		ob_start();
+		?>
+		<amp-story-page-attachment layout="nodisplay" theme="light" data-cta-text="<?php echo esc_attr( $attributes['text'] ); ?>" data-title="<?php echo esc_attr( $attributes['title'] ); ?>">
+			<div class="<?php echo esc_attr( $attributes['attachmentClass'] ); ?>" style="<?php echo esc_attr( $style ); ?>">
+				<h2><?php the_title(); ?></h2>
+				<?php the_content(); ?>
```

Comment:
Should this having a wrapping div here?

Reply:
You mean specifically the content?

Perhaps it makes more sense to iterate on this more broadly in a follow-up PR to allow overriding the markup via filters / theme templates.

Reply:
It just pretty normal to wrap `<div>` around content for styling reasons.

Reply:
Added one now.